opengrok-indexer/src/main/java/org/opengrok/indexer/web/messages/Message.java

```diff
@@ -183,11 +193,46 @@ public void serialize(
         public Duration deserialize(final JsonParser parser, final DeserializationContext context)
                 throws IOException {
             try {
+                // TODO Util.decode()
```

Comment:
The problem on the surface is that there is no `Util.decode()`. This could be resolved by replacing the encoding/decoding with `StringEscapeUtils` methods from Apache commons.

However, there is another problem that is related - due to the flattening in `AcceptedMessage` there is no deserialization done. The question how to work with user supplied content - should he get what he sent or just get the internal representation as it is currently.

Reply:
I think for Duration it's not a big deal.

I think we can do the internal representation as one day we might replace the static content in jsp with some javascript api calls to the messages endopint.

Reply:
That's what I had in mind when I was looking at the JS code.